### PR TITLE
Switch to FreezeUnfreeze for finetune

### PR DESCRIPTION
**Pull request by yawenzzzz**
1. Remove early stopping and previously lr scheduler, switch to FreezeUnfreeze and ReduceLROnPlateau. This setup is more similar to the finetuning recipe for our partner tasks, though training for less epochs, 50 instead of 100. 
2. Fix issue with AnySat, make sure only downsampling during train, not val/test.
3. Add patch size experiments for m_sa_crop_type and m_cashew_plant with OlmoEarth models.

> [!NOTE]
> Implements freeze–unfreeze finetuning with LR scheduling, threads is_train into embeddings/wrappers (fixing AnySat train-time subsampling), and trims/retunes finetune tasks and sweep configs.
> 
> - **Finetune**:
>   - Add freeze–unfreeze: freeze backbone for first 20% epochs, then unfreeze and drop LR 10×.
>   - Introduce `ReduceLROnPlateau`; remove per-step LR warmup/adjust and early stopping.
>   - Add `_set_backbone_trainable`; ensure eval uses `is_train=False`.
> - **Embeddings & Wrappers**:
>   - Thread `is_train` through `get_embeddings` and `EvalWrapper.__call__` (all subclasses).
>   - Replace `self.is_train` with `is_train` in `AnySatEvalWrapper` to control train-time pixel subsampling.
>   - Evaluator now passes `is_train` to `get_embeddings` and no longer sets it in wrapper ctor.
> - **Eval configs / sweep**:
>   - Streamline `FT_EVAL_TASKS` (comment out many; keep `m_sa_crop_type`, `m_cashew_plant` with batch/worker tweaks).
>   - Narrow LR sweep to `[1e-4, 5e-4, 1e-3]`; update run name suffixes to include `freezeunfreeze`; refresh example run strings.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 3a18ece5de3fc28d1d252dcb7807c64fe8ac5647. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: olmoearth_pretrain/train/callbacks/evaluator_callback.py [hunk 1]
```diff
@@ -324,22 +358,51 @@ def _val_finetune(self) -> tuple[float, float]:
...
+            logger.info(
+                f"Best checkpoint already exists for {self.evaluation_name}, skipping finetuning"
+            )
+            return 0.0, 0.0
+        else:
+            logger.info(
+                f"Best checkpoint does not exist for {self.evaluation_name}, running finetuning"
+            )
+
+        try:
```
Comment (favyen2) [inline olmoearth_pretrain/train/callbacks/evaluator_callback.py]: I don't think you need the try/except here. Anyway it will raise exception and exit if run_finetune_eval encounters error. Can just do like this:
```python
val_result, test_result = run_finetune_eval(...)
logger.info(...)
model.load_state_dict(original_state)
if original_training_mode:
    model.train()
else:
    model.eval()
```

File: olmoearth_pretrain/train/callbacks/evaluator_callback.py [hunk 2]
```diff
@@ -324,22 +358,51 @@ def _val_finetune(self) -> tuple[float, float]:
...
-            patch_size=self.patch_size,
-            pooling_type=self.pooling_type,
-            use_pooled_tokens=self.use_pooled_tokens,
-            train_loader=train_loader,
-            val_loader=val_loader,
-            test_loader=test_loader,
-        )
-        logger.info(
-            f"Downstream evaluator {self.evaluation_name} val score: {val_result}, test score: {test_result}"
+        # Check if best checkpoint already exists
```
Comment (Hgherzog) [inline olmoearth_pretrain/train/callbacks/evaluator_callback.py]: Suggestion: Move the checkpointing into a helper function

File: olmoearth_pretrain/train/callbacks/evaluator_callback.py [hunk 3]
```diff
@@ -312,38 +341,64 @@ def _val_finetune(self) -> tuple[float, float]:
...
+                model.eval()
 
-        torch.cuda.empty_cache()
-        gc.collect()
-        return val_result, test_result
+            torch.cuda.empty_cache()
+            gc.collect()
+            return val_result, test_result
 
     def val(self) -> tuple[float, float]:
```
Comment (Hgherzog) [inline olmoearth_pretrain/train/callbacks/evaluator_callback.py:405]: Why do we need to update the train vs eval state again after validation?

File: olmoearth_pretrain/evals/eval_wrapper.py [hunk 4]
```diff
@@ -50,7 +50,6 @@ def __init__(
         pooling_type: PoolingType,
         concat_features: bool = False,
         use_pooled_tokens: bool = False,
-        is_train: bool = True,
```
Comment (Hgherzog) [inline olmoearth_pretrain/evals/eval_wrapper.py:53]: Is this only used by anysat? Also, what was reason to move this out of the wrapper and into the method signature ?

File: olmoearth_pretrain/train/callbacks/evaluator_callback.py (hunk 1 above)
Comment (yawenzzzz) [reply olmoearth_pretrain/train/callbacks/evaluator_callback.py]: Fixed!

File: olmoearth_pretrain/train/callbacks/evaluator_callback.py [hunk 5]
```diff
@@ -312,38 +341,64 @@ def _val_finetune(self) -> tuple[float, float]:
...
+            self.evaluation_name,
+            f"lr{self.ft_lr}",
+            "best.ckpt",
         )
+        if os.path.exists(best_checkpoint_path):
+            logger.info("Best checkpoint already exists, skipping finetuning")
+            return 0.0, 0.0
+        else:
+            logger.info("Best checkpoint does not exist, running finetuning")
+            val_result, test_result = run_finetune_eval(
```
Comment (Hgherzog) [inline olmoearth_pretrain/train/callbacks/evaluator_callback.py:379]: When do we set model.eval() for running evaluations? I couldn't find it.

Comment (Hgherzog) [reply olmoearth_pretrain/train/callbacks/evaluator_callback.py:379]: Is it implicitly set in the saved checkpoint? I think it is better to explicitly set it or not before running evals.

File: olmoearth_pretrain/evals/eval_wrapper.py (hunk 4 above)
Comment (yawenzzzz) [reply olmoearth_pretrain/evals/eval_wrapper.py:53]: yes, this is only used by Anysat, the reason of moving it into the method signature is to avoid reinitialize the same wrapper twice, AnySat needs to do label sampling only in train, but using original label in val/test. with this, we can just pass the `is_train` to get different behaviors of label sampling or not.

File: olmoearth_pretrain/train/callbacks/evaluator_callback.py (hunk 5 above)
Comment (yawenzzzz) [reply olmoearth_pretrain/train/callbacks/evaluator_callback.py:379]: this is redundant, we already have `model.train()` and `module.eval()` in the `finetune.py`

Comment (yawenzzzz) [reply olmoearth_pretrain/train/callbacks/evaluator_callback.py:379]: i removed it.

File: olmoearth_pretrain/train/callbacks/evaluator_callback.py (hunk 3 above)
Comment (yawenzzzz) [reply olmoearth_pretrain/train/callbacks/evaluator_callback.py:405]: this is redundant, we already have `model.train()` and `module.eval()` in the `finetune.py`

File: olmoearth_pretrain/train/callbacks/evaluator_callback.py (hunk 2 above)
Comment (yawenzzzz) [reply olmoearth_pretrain/train/callbacks/evaluator_callback.py]: this is not checkpointing, but instead checking if best checkpoint exists or not, and the path will be passed to the `run_finetune_eval` function.  this is mainly used for resume, if best checkpoint exists, the task will be skipped.

Comment (Hgherzog) [reply olmoearth_pretrain/train/callbacks/evaluator_callback.py]: same nit applies, up to you

Comment (yawenzzzz) [reply olmoearth_pretrain/train/callbacks/evaluator_callback.py]: fixed.
